# this is the process I'm using to setup my first bioinformatics workflow from beginning to end before jumping over to our lab's remote computers.
```{r}
# get Rstudio and terminal aligned
Sys.setenv(RETICULATE_PYTHON = "/Users/cmantegna/miniforge3/envs/bio-cli/bin/python")
library(reticulate)
reticulate::py_config()

#also ensure it's added to RProfile - this is a one time thing and was done on 10.30.2025
```

```{bash}
# ideal repo setup
mussel_methylation/
├── data/
│   ├── raw/
│   └── processed/
├── code/
├── output/
├── r/
├── env/
│   └── environment.yml
├── .gitignore
├── mussel_methylation.Rproj
└── README.md

# need to add the env/ directory since it is my first time loading the conda/ mamba environment - this is code that is executed directly in terminal
mkdir env
# added the environment.yml file
# download the miniforge- conda packages to setup with the yml we created
# create the conda environment in terminal & restart terminal once complete
# activate and check versions of the programs you'll be using to verify everything loaded
# note: i want to leave the base deactivated most of the time if I'm running bioinfo projects, so i'll use "mamba activate bio-cli" when I need to use it and "mamba deactivate" when I'm done
# additionally, to make bio-cli visible in RStudio, I need to add it to my system settings.
# the path is: /Users/cmantegna/miniforge3/envs/bio-cli/bin/python
# update the .Rprofile file to include these lines and ensure everything is talking to each other.
# restart RStudio and verify that everything is working the next time 


```

```{bash}

mamba env create -f env/environment.yml
mamba activate bio-cli

```


